pkg/ipcache/ipcache.go

```diff
@@ -460,6 +468,31 @@ func (ipc *IPCache) UpsertMetadata(prefix netip.Prefix, src source.Source, resou
 	ipc.TriggerLabelInjection()
 }
 
+// UpsertMetadataBatch applies updates to multiple prefixes in a single transaction,
+// reducing potential lock contention.
+func (ipc *IPCache) UpsertMetadataBatch(updates []MetadataUpdate) {
```

Comment:
Hm, we could convert `[]MetadataUpdate` to be 

```
func (ipc *IPCache) UpsertMetadataBatch(updates ...MetadataUpdate) {
```

and that way in the future, we could have just `UpsertMetadata()` with a variadic arg and all the call sites would convert over to using `MetadataUpdate`. There would be no need to differentiate between a scalar call or a batch call because the variadic function can accept one or more args. In the latter case (via `len(update)`) the logic can detect a "batch" case and do the right thing.

I recognize that converting all the call sites to `MetadataUpdate` is likely out of scope for this PR, but making the API compatible with that change in this PR seems like a nice step.

Reply:
Done I also renamed the type to `MU`. Maybe that's OK.